events/src/camera.rs

```diff
@@ -1,15 +1,19 @@
 use serde::{Deserialize, Serialize};
+use url::Url;
 
 #[derive(Serialize, Deserialize, Debug, Clone)]
 pub struct Event {
     pub cameras: Vec<Camera>,
 }
 
 /// `Camera` type represents different camera types: IP, USB, CSI cameras
-#[derive(Serialize, Deserialize, Default, Debug, Clone)]
+#[derive(Serialize, Deserialize, Debug, Clone)]
 pub struct Camera {
-    /// ONVIF address if camera is ONVIF-compliant
-    pub onvif_addr: Option<String>,
+    /// URI to access the camera.
+    ///
+    /// - USB or CSI camera: local file name (for example `file:///dev/video0`).
+    /// - IP camera: ONVIF device management service URL (for example `http://192.168.1.2/device`);
+    pub uri: Url,
```

Comment:
You're also implicitly changing the type. Ideally that would be a separate change but you should mention it at least in the log.

Reply:
Yeah, will split into a separate commit when rebasing.